### PR TITLE
Fix issue 461: LoadModel misses textures

### PR DESCRIPTION
**Pull request by mingsong**
The root reason was the flaw of future textures that causes thread pool deadlocked when
doing async resource loading. This change addresses this directly when loading a model
from JassimpLoader by using callback mechanism instead of previous future texture pending
mechanism. A more fundermental redesign and implementation of the whole 
async resource loading infrastructure and thread scheduling is ongoing and coming soon, which will address all types of problems and performance issues we previously have seen.

### CODE REVIEW

**Comment by mingsong** [comment]
@rahul27 As I talked to you last Friday, there are two ways to bypass the problem in 461: one is to use sync loading in GVRJassimpSceneObject.java that runs the texture loading and  setting on glthread; and the other is like this change which uses callbacks instead of pending on future values.
Since async loading is error prone from previous experiences and performance also degrades some times, I am working on the comprehensive solution for the whole infrastructure. For now, this change will be enough to address 461.

**Comment by liaxim** [comment]
Looks good to me. @rahul27 Please confirm this resolves the problem you are experiencing.

**Comment by NolaDonato** [comment]
Mingsong,

I agree that using callbacks right now is the best way to fix this in the short term. SDC looms…

We can go back to using the Future infrastructure when you have stabilized it and it no longer deadlocks. Thanks for researching the asset loading and for finding a speedy fix.

-Nola

File: GVRf/Framework/src/org/gearvrf/jassimp2/GVRJassimpSceneObject.java [hunk 1]
```diff
@@ -145,15 +146,39 @@ private GVRSceneObject createSubSceneObject(
...
-                    Future<GVRTexture> futureDiffuseTexture = getGVRContext()
-                            .loadFutureTexture(resource);
-                    meshMaterial.setMainTexture(futureDiffuseTexture);
+                    
+                    TextureCallback callback = new TextureCallback() {
+                        @Override
+                        public void loaded(GVRTexture texture,
+                                GVRAndroidResource ignored) {
+                            meshMaterial.setMainTexture(texture);
+                            Log.e(TAG,
```
Comment (rahul27) [inline GVRf/Framework/src/org/gearvrf/jassimp2/GVRJassimpSceneObject.java]: Log.i?

**Comment by rahul27** [comment]
Verified using the sample app - It works.

I agree that we need to look into the threading long term because even with this fix I do see the occasional missing texture but am not able to reproduce it consistently (maybe a race condition). 

But for now now this should work.

**Pull request merged**